build-test.cmd

```diff
@@ -105,6 +105,7 @@ if defined __BuildAgainstPackagesArg (
 
 echo %__MsgPrefix%Using environment: "%__VSToolsRoot%\VsDevCmd.bat"
 call                                 "%__VSToolsRoot%\VsDevCmd.bat"
+@if defined _echo @echo on
```

Comment:
Is `_echo` already something the scripts respond to and it was missing these clauses, or is it something you're adding / used to debug?  What does it give us that isn't already in the log files?  This seems conceptually orthogonal to the issue you're fixing, and parameterizing shell scripts in new ways is something that might warrant a separate PR so it will be visible and discussion focused on it.  It doesn't bother me having this here, I just don't feel like the right person to sign off on it and worry that it could go under the radar since it's buried in the diffs of a PR that doesn't discuss it.

Reply:
This pattern, of turning echo back on after calling a script that turns it off (and when calling a script that ignores the `_echo` var), is already used in these scripts, but I added checks a few more places than before. It is for debugging the script. Otherwise, we get echoing, but only until calling the first script that turns it off.